public/app/plugins/panel/heatmap/heatmap_tooltip.ts

```diff
@@ -88,9 +88,17 @@ export class HeatmapTooltip {
     let tooltipTimeFormat = 'YYYY-MM-DD HH:mm:ss';
     let time = this.dashboard.formatDate(xData.x, tooltipTimeFormat);
 
-    let decimals = this.panel.tooltipDecimals || this.panelCtrl.decimals;
-    let scaledDecimals = this.panel.tooltipDecimals ? decimals - 2 : this.panelCtrl.scaledDecimals;
-    let valueFormatter = this.valueFormatter(decimals, scaledDecimals);
+    // Decimals override. Code from panel/graph/graph.ts
+    let valueFormatter;
+    if (_.isNumber(this.panel.tooltipDecimals)) {
+      valueFormatter = this.valueFormatter(this.panel.tooltipDecimals, null);
+    } else {
+      // auto decimals
+      // legend and tooltip gets one more decimal precision
+      // than graph legend ticks
+      let decimals = (this.panelCtrl.decimals || -1) + 1;
+      valueFormatter = this.valueFormatter(this.panel.tooltipDecimals, this.panelCtrl.scaledDecimals + 2);
```

Comment:
Is this a bug? It passes in this.panel.tooltipDecimals which is undefined instead of using the decimals variable.